### ISSUE TITLE
Unspecified class: Symbol

### ISSUE THREAD

**Issue by jellevandehaterd** [puppetlabs/puppetlabs-docker#715]
## Describe the Bug
Using module version 3.13.1 I get the following error when deploying a docker-compose.yml:

```
Info: Checking for compose project test
Error: /Stage[main]/Mycompany::Docker/Docker_compose[test]: Could not evaluate: Tried to load unspecified class: Symbol
```

## Expected Behavior
A successful check 

## Steps to Reproduce
Its not clear for me yet why this occurs but running 

`docker-compose -f docker-compose.yml -p test config`

and actually running the docker-compose.yml is no issue.
I do use a .env file next to the docker-compose.yml so I can use e.g. ${MY_VAR} in the compose file

## Environment
 - Version [3.13.1]
 - Platform [CentOS7]

## Additional Context
[YAML.safe_load](https://github.com/puppetlabs/puppetlabs-docker/blob/90725bb64b3254c5d69d0e18262f8d9f6c3b7c9d/lib/puppet/provider/docker_compose/ruby.rb#L23) probably is more strict than required.

My compose file:
```
version: "3.8"

services:
  test:
    image: traefik:v2.1.4
    logging:
      driver: "json-file"
      options:
        max-size: "50m"
        max-file: "7"
    networks:
      - traefik-public
      - mynetwork
    env_file:
      - ${TRAEFIK_ENV_FILE}
    ports:
      - 80:80
      - 443:443
      - 8080:8080
    volumes:
      - ${TRAEFIK_CONFIG_DIR}/:/config:ro
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /etc/pki/tls/private:/etc/ssl/certs:ro
    labels:
      traefik.enable: "true"
      traefik.http.routers.traefik.entrypoints: http
      traefik.http.routers.traefik.rule: "Host(\"my.host.name\")"
      traefik.http.routers.traefik.middlewares: traefik-https-redirect
      traefik.http.middlewares.traefik-https-redirect.redirectscheme.scheme: https
      traefik.http.routers.traefik-secure.entrypoints: https
      traefik.http.routers.traefik-secure.rule: "Host(\"my.host.name\")"
      traefik.http.routers.traefik-secure.tls: "true"
      traefik.http.routers.traefik-secure.tls.options: myTLSOptions@file
      traefik.http.routers.traefik-secure.service: api@internal
    restart: always

```

**Comment by loopiv** [comment]
Running into the same issue with mailu docker compose deployment (https://mailu.io/1.7/compose/setup.html).  If I modify lib/puppet/provider/docker_compose/ruby.rb and replace YAML.safe_load with YAML.load, things work as expected.  I'm not entirely sure what in the compose file is causing issues.  The compose file does use a env file.

**Comment by bjvrielink** [comment]
I am also running into this issue. I have tried different versions of this module, and can tell that the bug is introduced in 1.13.0 (t is not in 1.12.1).

I think the relevant change is that instead of parsing the docker-compose.yaml files, it now parses the output of 'docker-compose config'

In my docker-compose.yml I have:
   environment:
      - URL=http://some.host.name
      - ADDRESS=:9053

This is translated by docker-compose config to:
   environment:
      ADDRESS: :9053
      URL: http://some.host.name
And YAML.safe_load chokes on the :9053 token.

**Comment by Hvid** [comment]
I'm seeing the same thing with my compose for traefik:
```
TRAEFIK_ENTRYPOINTS_WEB_ADDRESS=:80
```
results in:
```
Could not evaluate: Tried to load unspecified class: Symbol
```

**Comment by linuxmail** [comment]
Hello,

same issue with the latest version from the module:

```
....
      - _REDIS_URL=:mysecret@redis:6379,100,
...
```

Fails, if I have i in the docker-compose.yml file. I have to remove / change it .. than it runs, but  on the next puppet run, it fails again.

**Comment by bmagistro** [comment]
In case others wind up here, it looks like this was fixed in v4.4.0 by #834 . I haven't had a chance to test yet, but the diff matches what I've changed locally.
